pkg/operator/controller/ingress/router_status.go

```diff
@@ -64,7 +64,8 @@ func (r *reconciler) isRouterDeploymentRolloutComplete(ic *operatorv1.IngressCon
 		return false, fmt.Errorf("failed to get deployment %s: %w", deploymentName, err)
 	}
 
-	if deployment.Status.Replicas != deployment.Status.UpdatedReplicas {
+	if deployment.Status.Replicas != deployment.Status.UpdatedReplicas ||
+		deployment.Generation != deployment.Status.ObservedGeneration {
 		return false, nil
 	}
```

Comment:
I think it's slightly stylistically better to use separate `if` statements because the conditions aren't integrally related, and avoiding line-wrapping is good:
```suggestion
	if deployment.Generation != deployment.Status.ObservedGeneration {
		return false, nil
	}
	if deployment.Status.Replicas != deployment.Status.UpdatedReplicas {
		return false, nil
	}
```